src/viam/app/data_client.py

```diff
@@ -269,7 +269,11 @@ async def tabular_data_by_sql(self, organization_id: str, sql_query: str) -> Lis
         response: TabularDataBySQLResponse = await self._data_client.TabularDataBySQL(request, metadata=self._metadata)
         return [bson.decode(bson_bytes) for bson_bytes in response.raw_data]
 
-    async def tabular_data_by_mql(self, organization_id: str, mql_binary: List[bytes]) -> List[Dict[str, Union[ValueTypes, datetime]]]:
+    async def tabular_data_by_mql(
+        self,
+        organization_id: str,
+        mql_query: Union[List[bytes], Dict[str, Union[ValueTypes, datetime]]]
```

Comment:
The parameter name change from `mql_binary` to `mql_query` is questionable. Even though we only want to accept queries, we wanted to make a union so that it wasn't a breaking change yet. Changing the name, however, is a breaking change to any user using named parameters, so at this point, one could argue we might as well change the input type.
On the other hand, leaving it as `mql_binary` is now incorrect, and might push users to keep using bytes when we don't want that. Fine with either way, as long as it's in the direction we want the SDK to go.